services/src/main/java/org/exoplatform/gamification/connectors/github/services/GithubHooksManagement.java

```
@@ -0,0 +1,264 @@
+            os.flush();
+        } catch (IOException e) {
+            LOG.error(e);
+        }
+        int responseCode = patchConnection.getResponseCode();
+        if (responseCode == HttpURLConnection.HTTP_OK) { //success
+            return responseCode;
+        } else {
+            try (BufferedReader in = new BufferedReader(new InputStreamReader(patchConnection.getErrorStream()))){
+                String inputLine;
+                StringBuffer response = new StringBuffer();
+                while ((inputLine = in.readLine()) != null) {
+                    response.append(inputLine);
+                }
+                ObjectMapper objectMapper = new ObjectMapper();
+                JsonNode infoNode = objectMapper.readTree(response.toString());
+                String errMessage = infoNode.get("message").textValue();
+                if (infoNode.get("errors") != null) {
+                    errMessage = errMessage + ": " + infoNode.get("errors").elements().next().get("message").textValue();
+                }
+                throw new GithubHookException(errMessage);
+            } catch (Exception e) {
+                LOG.error(e);
+            }
+        }
+        return 0;
+    }
+
+
+    public  int deleteHook(long webhook, String org, String repo) throws IOException, GithubHookException {
+        JSONObject config = new JSONObject();
+        JSONObject hook = new JSONObject();
+        String url = "https://api.github.com/repos/" + org + "/" + repo + "/hooks/" + webhook;
+
+        URL obj = new URL(url);
+        HttpURLConnection deleteConnection = (HttpURLConnection) obj.openConnection();
+        deleteConnection.setRequestMethod("DELETE");
+        deleteConnection.setRequestProperty("Authorization", "token " + TOKEN);
+        int responseCode = deleteConnection.getResponseCode();
+        if (responseCode == HttpURLConnection.HTTP_NO_CONTENT) { //success
+            return responseCode;
```

It should throw an exception instead of returning an integer.